internal/internal_task_handlers.go

```diff
@@ -616,6 +616,10 @@ func (wth *workflowTaskHandlerImpl) ProcessWorkflowTask(
 
 	workflowContext, err := wth.getOrCreateWorkflowContext(task, historyIterator)
 	if err != nil {
+		// The WorkflowContext will be removed from the cache if an error is found
+		// in workflowContext.err when calling Unlock() method.
+		workflowContext.err = err
+		defer workflowContext.Unlock()
```

Comment:
I think here is what we need to update. Change it to:
defer func() { workflowContext.Unlock(err); }
and evict the context within the Unlock method when err is not nil. You also need to change the signature of Unlock().

Reply:
will do. 
Thanks!